### PR TITLE
elevation calculation fix

### PR DESCRIPTION
**Pull request by angelk**
Fix for https://github.com/trackhub/website/issues/188

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 dist: bionic
 language: php
 php:
-  - 7.3
+  - 7.4
```
Comment (StefanMavrodiev) [inline .travis.yml]: IMHO, this should be separate PR.

File: composer.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
     "type": "project",
     "license": "proprietary",
     "require": {
-        "php": ">=7.3",
+        "php": ">=7.4",
```
Comment (StefanMavrodiev) [inline composer.json]: Same here.

File: src/Track/Processor.php [hunk 3]
```diff
@@ -14,11 +15,21 @@ class Processor
...
+     * @var ElevationNoiseReducerInterface[]
+     */
+    private $elevationNoceReducer = [];
+
     public function addTwoPointsChecker(TwoPointsChecker\PointCheckerInterface $checker)
     {
         $this->twoPointsCheckers[] = $checker;
     }
 
+    public function addElevationNoiseReducer(ElevationNoiseReducerInterface $reductor)
```
Comment (StefanMavrodiev) [inline src/Track/Processor.php]: Can we name this variable $reducer?

File: src/Track/Processor.php [hunk 4]
```diff
@@ -93,23 +106,22 @@ public function process(string $source, Version $version)
...
+                                $elevationChange = $point->getElevation() - $previousElevationPoint->getElevation();
                                 if ($previousRealPoint === null || $this->isPointReal($previousRealPoint, $point)) {
-                                    $positiveElevation += $elevationChange;
                                     $previousRealPoint = $point;
                                 } else {
                                     $point->setElevation();
                                 }
                             } else {
-                                $elevationChange = $previousElevation - $point->getElevation();
+                                $elevationChange = $previousElevationPoint->getElevation() - $point->getElevation();
```
Comment (StefanMavrodiev) [inline src/Track/Processor.php]: Is it possible *$previousElevationPoint* to be null here? E.g. *previousPoint* is set set, but *getElevation* returns null?

File: .travis.yml (hunk 1 above)
Comment (angelk) [reply .travis.yml]: reverted

File: composer.json (hunk 2 above)
Comment (angelk) [reply composer.json]: reverted

File: src/Track/Processor.php (hunk 3 above)
Comment (angelk) [reply src/Track/Processor.php]: fixed

File: src/Track/Processor.php (hunk 4 above)
Comment (angelk) [reply src/Track/Processor.php]: No, previousElevationPoint always have elevation, see https://github.com/trackhub/website/pull/198/files#diff-88877c04d979dfdbd6321a823af1462fR138
